plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftCommandLineToolsRequirement.java

```diff
@@ -71,75 +85,88 @@ public void setDeclaration(OCBinary declaration) {}
 		logger.info("Creating directory binaries");
-		FileHelper.createDirectory(new File("binaries"));
-		URL downloadLink = null;
+		File outputDirectory = new File(CLIENT_TOOLS_DESTINATION);
+		FileHelper.createDirectory(outputDirectory );
+
+		String fileName = downloadArchive(getDownloadLink());
+		String extractedDirectory = extractArchive(fileName, outputDirectory);
+
+		if (StringUtils.isEmpty(extractedDirectory)
+			|| !(new File(extractedDirectory).exists())) {
+				throw new OpenShiftToolsException("Cannot extract archive " + fileName + ". "
+						+ "Archive does not extract into a single root folder.");
+		}
+
+		return new File(extractedDirectory, OCBinaryFile.get().getName());
+	}
+
+	private String downloadArchive(String downloadLink) {
+		if (StringUtils.isEmpty(downloadLink)) {
+			throw new OpenShiftToolsException("Cannot download OpenShift binary. No download known\n");
+		}
+
 		String fileName = null;
-		FileOutputStream fileOutputStream = null;
-		ReadableByteChannel readableByteChannel = null;
-		
 		try {
-			logger.info("Processing download link.");
-			downloadLink = new URL(getDownloadLink());
-			fileName = getFileName(downloadLink.getPath());
-			
-			logger.info("Downloading OpenShift binary.");
-			readableByteChannel = Channels.newChannel(downloadLink.openStream());
-			fileOutputStream = new FileOutputStream(fileName);
-			fileOutputStream.getChannel().transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
-		} catch (MalformedURLException ex) {
-			// Never thrown because links are valid
-		} catch (IOException ex) {
-			throw new OpenShiftToolsException("Cannot download OpenShift binary.\n" + ex.getMessage());
-		} finally {
-			try {
-				readableByteChannel.close();
-			} catch (IOException ex) {
-				// do nothing
+			URL downloadUrl = new URL(downloadLink);
+			fileName = getFileName(downloadUrl.getPath());
+			if (new File(fileName).exists()) {
+				return fileName;
 			}
-			try {
-				fileOutputStream.close();
-			} catch (Exception ex) {
-				// do nothing
+			try (FileOutputStream fileOutputStream = new FileOutputStream(fileName);
+					ReadableByteChannel readableByteChannel = Channels.newChannel(downloadUrl.openStream())) {
+				logger.info("Downloading OpenShift binary.");
+				fileOutputStream.getChannel().transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
+			} catch (IOException ex) {
+				throw new OpenShiftToolsException("Cannot download OpenShift binary.\n" + ex.getMessage());
 			}
+		} catch (MalformedURLException e) {
+			throw new OpenShiftToolsException(NLS.bind("Could not download \"{0}\". Invalid url.", downloadLink));
 		}
+		return fileName;
+	}
 
-		logger.info("Extracting OpenShift archive.");
-		File outputDirectory = new File("binaries");
-		if (fileName.endsWith("zip")) {
+	private String extractArchive(String fileName, File outputDirectory) {
+		if (StringUtils.isEmpty(fileName)) {
+			return null;
+		}
+
+		logger.info("Extracting OpenShift archive {}.", fileName);
```

Comment:
missing number of argument {1} thus logger output is "Extracting OpenShift archive {}." instead of "Extracting OpenShift archive archive_file_name.tar.gz."